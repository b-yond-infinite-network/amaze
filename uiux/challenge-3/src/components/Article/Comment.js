import DeleteButton from './DeleteButton';
import { Link } from 'react-router-dom';
import React from 'react';

const Comment = props => {
  const comment = props.comment;
  const show = props.currentUser &&
    props.currentUser.username === comment.author.username;
  return (
    <div id="comment-card" className="card">
      <div className="card-block">
        <p className="card-text">{comment.body}</p>
      </div>
      <div className="card-footer">
        <Link
        id="comment-author-image"
          to={`/@${comment.author.username}`}
          className="comment-author">
          <img src={comment.author.image} className="comment-author-img" alt={comment.author.username} />
        </Link>
        &nbsp;
        <Link
        id="comment-author"
          to={`/@${comment.author.username}`}
          className="comment-author">
          {comment.author.username}
        </Link>
        <span id="comment-created" className="date-posted">
          {new Date(comment.createdAt).toDateString()}
        </span>
        <DeleteButton id="comment-delete-button" show={show} slug={props.slug} commentId={comment.id} />
      </div>
    </div>
  );
};

export default Comment;
