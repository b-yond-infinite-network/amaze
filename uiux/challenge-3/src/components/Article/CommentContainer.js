import CommentInput from './CommentInput';
import CommentList from './CommentList';
import { Link } from 'react-router-dom';
import React from 'react';

const CommentContainer = props => {
  if (props.currentUser) {
    return (
      <div id="comment-container" className="col-xs-12 col-md-8 offset-md-2">
        <div>
          <list-errors errors={props.errors}></list-errors>
          <CommentInput id="comment-input" slug={props.slug} currentUser={props.currentUser} />
        </div>

        <CommentList
          id="comment-list"
          comments={props.comments}
          slug={props.slug}
          currentUser={props.currentUser} />
      </div>
    );
  } else {
    return (
      <div id="comment-container" className="col-xs-12 col-md-8 offset-md-2">
        <p>
          <Link id="comment-login" to="/login">Sign in</Link>
          &nbsp;or&nbsp;
          <Link id="comment-register" to="/register">sign up</Link>
          &nbsp;to add comments on this article.
        </p>

        <CommentList
          id="comment-list"
          comments={props.comments}
          slug={props.slug}
          currentUser={props.currentUser} />
      </div>
    );
  }
};

export default CommentContainer;
