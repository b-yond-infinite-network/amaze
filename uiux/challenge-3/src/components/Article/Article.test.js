import Article from './Article';
import { shallow, mount } from 'enzyme';
import React from 'react';
import renderer from 'react-test-renderer'
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router'
import configureMockStore from 'redux-mock-store'
import middlewares from '../../store'

const state = {
    common: {
        currentUser: {
            username: "test username"
        }
    },
    match: {
        params: {
            id: '1234'
        }
    },
    currentUser: {
        username: "test username"
    },
    article: {
        title: 'Untitled Title',
        slug: 'a-title-7ua6fd',
        body: '# Nic\nbylo nebylo',
        createdAt: '2019-06-29T18:36:23.216Z',
        updatedAt: '2019-06-29T18:37:11.601Z',
        tagList: [
            'nothing',
            'nic'
        ],
        description: 'o ničem',
        author: {
            username: 'tester96',
            bio: 'b',
            image: '',
            following: false
        },
        favorited: false,
        favoritesCount: 1
    },
    comments: [],
    commentErrors: []
};

describe('Article Component', () => {

    let mockStore;

    let tree;
    let wrapper;
    let render;

    beforeAll(() => {

        mockStore = configureMockStore(middlewares)(state);

        const props = state;

        tree = <Provider store={mockStore}>
            <MemoryRouter initialEntries={["/"]}>
                <Article {...props} />
            </MemoryRouter>
        </Provider>
        wrapper = mount(tree);
        console.log(wrapper.debug());
        render = renderer.create(tree).toJSON();

    });

    it('Should match the snapshot', () => {
        expect(render).toMatchSnapshot();
    });

    it('Should render without errors', () => {
        const component = wrapper.find('#article-container');
        expect(component.length).toBe(1);
    });

});