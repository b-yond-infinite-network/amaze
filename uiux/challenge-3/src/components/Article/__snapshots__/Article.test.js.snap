// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Article Component Should match the snapshot 1`] = `
<div
  className="article-page"
  id="article-container"
>
  <div
    className="banner"
  >
    <div
      className="container"
    >
      <h1>
        Untitled Title
      </h1>
      <div
        className="article-meta"
        id="article-meta"
      >
        <a
          href="/@tester96"
          id="article-meta-author-image"
          onClick={[Function]}
        >
          <img
            alt="tester96"
            src=""
          />
        </a>
        <div
          className="info"
        >
          <a
            className="author"
            href="/@tester96"
            id="article-meta-author-info"
            onClick={[Function]}
          >
            tester96
          </a>
          <span
            className="date"
            id="article-meta-created"
          >
            Sat Jun 29 2019
          </span>
        </div>
        <span />
      </div>
    </div>
  </div>
  <div
    className="container page"
  >
    <div
      className="row article-content"
    >
      <div
        className="col-xs-12"
      >
        <div
          dangerouslySetInnerHTML={
            Object {
              "__html": "<h1 id=\\"nic\\">Nic</h1>
<p>bylo nebylo</p>
",
            }
          }
        />
        <ul
          className="tag-list"
        >
          <li
            className="tag-default tag-pill tag-outline"
          >
            nothing
          </li>
          <li
            className="tag-default tag-pill tag-outline"
          >
            nic
          </li>
        </ul>
      </div>
    </div>
    <hr />
    <div
      className="article-actions"
    />
    <div
      className="row"
    >
      <div
        className="col-xs-12 col-md-8 offset-md-2"
        id="comment-container"
      >
        <div>
          <list-errors
            errors={Array []}
          />
          <form
            className="card comment-form"
            onSubmit={[Function]}
          >
            <div
              className="card-block"
            >
              <textarea
                className="form-control"
                onChange={[Function]}
                placeholder="Write a comment..."
                rows="3"
                value=""
              />
            </div>
            <div
              className="card-footer"
            >
              <img
                alt="test username"
                className="comment-author-img"
              />
              <button
                className="btn btn-sm btn-primary"
                type="submit"
              >
                Post Comment
              </button>
            </div>
          </form>
        </div>
        <div />
      </div>
    </div>
  </div>
</div>
`;
